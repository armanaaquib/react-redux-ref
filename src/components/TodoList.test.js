import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Todo from './Todo';
import TodoList from './TodoList';

describe('TodoList', () => {
  const todos = [
    {
      id: 1,
      text: 'Consider using Redux',
      completed: true,
    },
    {
      id: 2,
      text: 'Keep all state in a single tree',
      completed: false,
    },
  ];

  it('should match snapshot', () => {
    const wrapper = shallow(<TodoList todos={todos} toggleTodo={jest.fn()} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass all required props to Todo', () => {
    const mockOnTodoClick = jest.fn();
    const wrapper = mount(
      <TodoList todos={todos} toggleTodo={mockOnTodoClick} />
    );

    const props = wrapper.find(Todo).first().props();

    expect(props).toHaveProperty('id', 1);
    expect(props).toHaveProperty('completed', true);
    expect(props).toHaveProperty('text', 'Consider using Redux');
  });

  it('should call togllTodo with todo id', () => {
    const mockOnTodoClick = jest.fn((id) => {
      expect(id).toBe(1);
    });

    const wrapper = shallow(
      <TodoList todos={todos} toggleTodo={mockOnTodoClick} />
    );

    wrapper.find(Todo).first().simulate('click');
  });
});
