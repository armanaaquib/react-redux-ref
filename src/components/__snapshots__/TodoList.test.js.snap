// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match snapshot 1`] = `
<ul>
  <Todo
    completed={true}
    id={1}
    key="1"
    onClick={[Function]}
    text="Consider using Redux"
  />
  <Todo
    completed={false}
    id={2}
    key="2"
    onClick={[Function]}
    text="Keep all state in a single tree"
  />
</ul>
`;
